### ISSUE TITLE
defintion for rule was not found

### ISSUE THREAD

**Issue by PradheepGit** [eslint/eslint#12146]
**Tell us about your environment**

* **ESLint Version:6.2.1
* **Node Version:10.16.0
* **npm Version:6.9.0

**What parser (default, Babel-ESLint, etc.) are you using?**

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
```

```bash
eslint --no-eslintrc --rule "alert:2" --rulesdir ./ ~/Downloads/Eclipse.app/Cont
ents/MacOS/ZIDE/javascript/templateInner.js
```

**What did you expect to happen?**
I expected that ESlint would report only the errors related to the rule but it reports multiple 'definition for rule was not found' along with that error.

**What actually happened? Please include the actual, raw output from ESLint.**

![image](https://user-images.githubusercontent.com/54398787/63507609-c143fa00-c4f5-11e9-9fbe-19721b992907.png)
As my 'rulesdir' has no rule except the above mentioned rule.
I hope something is wrong with my command
**Are you willing to submit a pull request to fix this bug?**
No

**Comment by ilyavolodin** [comment]
Does `./` contains any rules? `--rulesdir` will automatically load them. But it does seem strange that `--no-eslintrc` would not completely clear your configuration. Could you try using `--print-config` flag and see if generated config contains the rules that are not found? You can also use `--debug` flag to see which config files are being accessed.

**Comment by PradheepGit** [comment]
yeah ./ contains the single rule 'alert' so i got confused. sure  i will
try the suggested commands.

**Comment by PradheepGit** [comment]
Config using --print-config 
![image](https://user-images.githubusercontent.com/54398787/63573378-c9a63e80-c5a2-11e9-978b-0c3ecd90ebce.png)

**Comment by ilyavolodin** [comment]
This is very strange. Would it be possible to somehow create a basic repository with reproduction steps? This shouldn't be happening, but I don't know what might be causing this at all.

**Comment by PradheepGit** [comment]
But with older Version (5.16.0) it's working fine

**Comment by PradheepGit** [comment]
Working fine for the same file with version (5.16.0)
![image](https://user-images.githubusercontent.com/54398787/63599410-2d4e5d00-c5df-11e9-9421-06eb4c76ec24.png)
